src/api/server/server.js

```diff
@@ -1,33 +1,45 @@
 'use strict';
 const common = require('../common');
 
+// If a ledger is not received in this time, consider the connection offline
+const CONNECTION_TIMEOUT = 1000 * 30;
+
 function connect(callback) {
   this.remote.connect(callback);
 }
 
+function disconnect(callback) {
+  this.remote.disconnect(callback);
+}
+
+function isUpToDate(remote) {
+  const server = remote.getServer();
+  return (server ? true : false) && (remote._stand_alone
+    || (Date.now() - server._lastLedgerClose) <= CONNECTION_TIMEOUT);
+}
+
 function isConnected() {
-  return common.server.isConnected(this.remote);
+  return !isNaN(this.remote._ledger_current_index) && isUpToDate(this.remote);
```

Comment:
`0` is not a valid ledger index, and is `falsy` along with undefined ( the default value ). I'd probably just go with `this.remote._ledger_current_index && $expr`.  `Not is Not A Number` is mental mouthful.